cosmwasm/contracts/crosschain-swaps/src/execute.rs

```diff
@@ -34,13 +35,16 @@ pub fn swap_and_forward(
     deps.api.debug(&format!("executing swap and forward"));
     let config = CONFIG.load(deps.storage)?;
 
+    // TODO: Here we can take the fees
+    let swap_coin = Coin::new(swap_amount, received_coin.denom);
```

Comment:
I mean why would we not use `received_coin.amount` here? What is the reason for having `swap_amount` separately when `received_coin`'s amount is unused

Reply:
I'm down to not block on this and merge the PR if my proposal seems undesirable

Reply:
using `received_coin.amount` there is the same as just using `received_coin`. That would work right now. 

The reson to have it be different is that I'm planning changes to the contract that will require the amount you send to be slightly higher than the swapped amount. Specifically, fees. If we want to charge a fee for the transfer, the user needs to send swap_amount+fee_amount to the contract. So, if they want to swap 10osmo for juno (assuming 1:1 price and a fee of 1%) they will have to send 11 osmo to the contract so they pay 1 osmo in fees and get 10 juno back. To specify that, the send 11 osmo, but in the message include `"swap_amount": "10"`.

Since the fees may not be known in advance, I'm also planning a "cash-back" mechanism where they would be allowed to send a much higher amount without loss: send 100 osmo to the contract (with swap_amount=10), and get back: 10juno, 89osmo (where 1 osmo went to pay the fee). 

(In the last example the fee is specified as 1 osmo and not as a percentage. Still need to design the best strategy for fees)